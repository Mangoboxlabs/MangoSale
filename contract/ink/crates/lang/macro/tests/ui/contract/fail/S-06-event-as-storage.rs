use ink_lang as ink;

#[ink::contract]
mod event_as_storage {
    #[ink(event)]
    #[ink(storage)] // We cannot have #[ink(storage)] if we already have #[ink(event)]
    pub struct EventAsStorage {}

    impl EventAsStorage {
        #[ink(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        pub fn message(&self) {}
    }
}

fn main() {}
