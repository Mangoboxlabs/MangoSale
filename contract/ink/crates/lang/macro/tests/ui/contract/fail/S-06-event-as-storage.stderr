error: encountered conflicting ink! attribute argument
 --> $DIR/S-06-event-as-storage.rs:6:11
  |
6 |     #[ink(storage)] // We cannot have #[ink(storage)] if we already have #[ink(event)]
  |           ^^^^^^^
